ModelLibrary/Process/CalculateCrossCurrRate.cs

```diff
@@ -0,0 +1,183 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using VAdvantage.Model;
+using VAdvantage.DataBase;
+using System.Data;
+using VAdvantage.Logging;
+using VAdvantage.Utility;
+using VAdvantage.ProcessEngine;
+namespace VAdvantage.Process
+{
+    public class CalculateCrossCurrRate : SvrProcess
+    {
+        StringBuilder retmsg = new StringBuilder();
+        StringBuilder query = new StringBuilder();
+        string name = string.Empty;
+        int Client_ID = 0;
+        int Org_ID = 0;
+        int CurrType = 0;
+        int CommonCurr = 0;
+        int FromCurr = 0;
+        int ToCurr = 0;
+        int FromCurr1 = 0;
+        DateTime ValidFrom1 = new DateTime();
+        DateTime ValidFrom2 = new DateTime();
+        DateTime ValidTo1 = new DateTime();
+        DateTime ValidTo2 = new DateTime();
+        Decimal newMulRate = 0;
+        Decimal newDivideRate = 0;
+        Decimal MulRate1, MulRate2 = 0;
+        DateTime newValidFrom = new DateTime();
+        DateTime newValidTo = new DateTime();
+
+        protected override string DoIt()
+        {
+            //string sysDate = System.DateTime.Now.Date;
+             string sysDate = "trunc(sysdate)";
+            if (DB.IsPostgreSQL())
+            {
+                sysDate = "current_timestamp";
+            }
+            // Getting records from cross rate setting
+            query.Append("SELECT * FROM C_CurrCrossRate WHERE IsActive='Y'");
+            DataSet dsobj = DB.ExecuteDataset(query.ToString());
+            query.Clear();
+            if (dsobj != null && dsobj.Tables.Count > 0 && dsobj.Tables[0].Rows.Count > 0)
+            {
+                for (int i = 0; i < dsobj.Tables[0].Rows.Count; i++)
+                {
```

Comment:
Take DataRow and create Object of MClass using that, then use that object to get values